testsuite/smoke-test/src/execution_pool.rs

```diff
@@ -0,0 +1,107 @@
+    utils::{update_consensus_config, MAX_CATCH_UP_WAIT_SECS},
+};
+use aptos::test::CliTestFramework;
+use aptos_forge::{LocalSwarm, NodeExt, Swarm, SwarmExt};
+use aptos_rest_client::Client;
+use aptos_types::on_chain_config::{
+    ConsensusAlgorithmConfig, OnChainConsensusConfig, ValidatorTxnConfig,
+};
+use move_core_types::language_storage::CORE_CODE_ADDRESS;
+use std::{sync::Arc, time::Duration};
+use tokio::task::JoinHandle;
+
+/// Checks the value of the `window_size` in the [`OnChainConsensusConfig`](OnChainConsensusConfig)
+pub async fn assert_on_chain_consensus_config_window_size(
+    swarm: &mut LocalSwarm,
+    expected_window_size: Option<usize>,
+) {
+    let rest_client = swarm.validators().next().unwrap().rest_client();
+    let current_consensus_config: OnChainConsensusConfig = bcs::from_bytes(
+        &rest_client
+            .get_account_resource_bcs::<Vec<u8>>(
+                CORE_CODE_ADDRESS,
+                "0x1::consensus_config::ConsensusConfig",
+            )
+            .await
+            .unwrap()
+            .into_inner(),
+    )
+    .unwrap();
+
+    match current_consensus_config {
+        OnChainConsensusConfig::V1(_)
+        | OnChainConsensusConfig::V2 { .. }
+        | OnChainConsensusConfig::V3 { .. } => {
+            panic!("Expected OnChainConsensusConfig::V4, but received a different version")
+        },
+        OnChainConsensusConfig::V4 { window_size, .. } => {
+            assert_eq!(window_size.map(|v| v as usize), expected_window_size)
+        },
+    }
+}
+
+async fn initialize_swarm_with_window() -> (
+    LocalSwarm,
+    CliTestFramework,
+    JoinHandle<anyhow::Result<()>>,
+    usize,
+    Client,
+) {
+    let (swarm, mut cli, _faucet) = SwarmBuilder::new_local(4)
+        .with_init_config(Arc::new(|_, conf, _| {
+            // reduce timeout, as we will have dead node during rounds
+            conf.consensus.round_initial_timeout_ms = 400;
+            conf.consensus.quorum_store_poll_time_ms = 100;
+            conf.api.failpoints_enabled = true;
+        }))
+        .with_init_genesis_config(Arc::new(|genesis_config| {
+            genesis_config.consensus_config = OnChainConsensusConfig::V4 {
+                alg: ConsensusAlgorithmConfig::default_for_genesis(),
+                vtxn: ValidatorTxnConfig::default_for_genesis(),
+                window_size: Some(4u64),
+            };
+        }))
+        .with_aptos()
+        .build_with_cli(0)
+        .await;
+
+    let root_cli_index = cli.add_account_with_address_to_cli(
+        swarm.root_key(),
+        swarm.chain_info().root_account().address(),
+    );
+
+    let rest_client = swarm.validators().next().unwrap().rest_client();
+
+    (swarm, cli, _faucet, root_cli_index, rest_client)
+}
+
+#[tokio::test]
+async fn test_window_size_onchain_config_change() {
+    let window_size = Some(4usize);
+    let (mut swarm, cli, _faucet, root_cli_index, ..) = initialize_swarm_with_window().await;
```

Comment:
it's a little awkward that the window size initialized is a magic number inside, probably pass it in and assert?